### PR TITLE
Cumulus 817 dead letter queues

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-817: Add dead letter queue configuration/](https://bugs.earthdata.nasa.gov/browse/CUMULUS-817)

## Changes

* Adds configuration option `namedLambdaDeadLetterQueue` to custom compilation templates that adds a SQS DLQ to the template and sets the configured lambda to target it.  

## Test Plan
Things that should succeed before merging.

- [x] Unit tests
- [x] Adhoc testing
- [x] Update CHANGELOG

### CODE REVIEW

File: packages/api/config/lambdas.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ sqs2sf:
   memory: 512
   source: 'node_modules/@cumulus/api/dist/'
```
Comment (laurenfrederick) [inline packages/api/config/lambdas.yml]: I'm thinking this one might need a dlq?

File: packages/api/config/lambdas.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ sqs2sf:
...
 
+
 executeMigrations:
```
Comment (laurenfrederick) [inline packages/api/config/lambdas.yml]: What did we decide on this one?

**Comment by laurenfrederick** [comment]
Oh also I looked into if there is a limit on how many queues you can have on your stack and didn't see anything, so I don't anticipate running into problems with limits.

File: packages/deployment/app/config.yml [hunk 3]
```diff
@@ -13,6 +13,13 @@ default:
   message_adapter_repo: cumulus-message-adapter
   message_adapter_filename: cumulus-message-adapter.zip
 
+  customCompilation: true
+  injectWorkflowLambdaAliases: true
+
+  DLQDefaultTimeout: 60
+  DLQDefaultMessageRetentionPeriod: 1209600
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply packages/deployment/app/config.yml]: 👍  good point.

File: packages/api/config/lambdas.yml (hunk 1 above)
Comment (Jkovarik) [reply packages/api/config/lambdas.yml]: I believe this one explicitly does not, as the SQS queue it's pulling from via the starSFWatcherRule has it's own targeted DLQ -- this allows that queue to detect if this lambda fails and puts the message in startSFFailedSQS for reprocessing.     I have a brief note on this in the resource survey on the team drive.

File: packages/api/config/lambdas.yml (hunk 2 above)
Comment (Jkovarik) [reply packages/api/config/lambdas.yml]: We should chat about this one, it's tied up in the async operations/Bulk Delete work that isn't completed.

File: packages/deployment/app/config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jkovarik) [reply packages/deployment/app/config.yml]: b7bb83e7

File: packages/api/config/lambdas.yml (hunk 2 above)
Comment (Jkovarik) [reply packages/api/config/lambdas.yml]: I've posed a question about this ticket's AC to @bilts, we can always add the config option if it's determined my assertion is incorrect prior to closing 817.

**Pull request merged**